### PR TITLE
bugfix/21405-stickytracking-not-working-for-bubbleseries

### LINKED ISSUE #21405: Issue with hovering over overlapping bubbles with sticky tracking disabled.
#### Expected behaviour
If sticky tracking is disabled, I expect that the bubble that the mouse is actually on will be hovered.

#### Actual behaviour
Sometimes a bubble that is close to the mouse pointer is hovered, but not the one over which the mouse actually is.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/nL20xoae/

#### Product version
Highcharts v11.4.3
Last working version: v11.1.0

#### Workaround
Custom events with `enableMouseTracking` disabled:
https://jsfiddle.net/BlackLabel/1cL68x2d/

### PR DESCRIPTION
**Pull request by MarkusBarstad**
Fixed #21405, `stickyTracking` was broken for `BubbleSeries`.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207666203929272

### CODE REVIEW

File: ts/Core/Series/Series.ts [hunk 1]
```diff
@@ -3666,6 +3666,7 @@ class Series {
             kdDimensions = (series.options.findNearestPointBy || '')
                 .indexOf('y') > -1 ? 2 : 1,
             useRadius = !!series.isBubble;
+            // GetDist = (x, y) => Math.sqrt(x * x, y * y);
```
Comment (bre1470) [inline ts/Core/Series/Series.ts:3669]: Remove?

File: ts/Core/Series/Series.ts [hunk 2]
```diff
@@ -3722,6 +3723,8 @@ class Series {
                         nPoint1 :
                         point
                 );
+
+                // }
```
Comment (bre1470) [inline ts/Core/Series/Series.ts:3727]: Remove?

File: ts/Series/PackedBubble/PackedBubblePoint.ts [hunk 3]
```diff
@@ -157,7 +157,7 @@ class PackedBubblePoint extends BubblePoint implements DragNodesPoint {
  *
  * */
 
-interface PackedBubblePoint extends NetworkgraphPoint {
+interface PackedBubblePoint extends Omit<NetworkgraphPoint, 'onMouseOver' | 'optionsToObject'> {
```
Comment (bre1470) [inline ts/Series/PackedBubble/PackedBubblePoint.ts:160]: Please keep it simple. If it doesn't match then it is not extending.

File: ts/Core/Series/Series.ts [hunk 4]
```diff
@@ -3683,7 +3684,7 @@ class Series {
                 y = (defined(p1kdY) && defined(p2kdY)) ? p1kdY - p2kdY : 0,
                 radius = useRadius ? (p2.marker?.radius || 0) : 0;
 
-            p2.dist = Math.sqrt(((x && x * x) || 0) + y * y) - radius;
+            p2.dist = Math.abs(Math.sqrt(((x && x * x) || 0) + y * y) - radius);
```
Comment (bre1470) [inline ts/Core/Series/Series.ts:3687]: Nice and simple.

**Review by bre1470 — CHANGES_REQUESTED** [review]
Solution looks good. 👍  Adjustments needed though.

File: ts/Series/PackedBubble/PackedBubblePoint.ts (hunk 3 above)
Comment (MarkusBarstad) [reply ts/Series/PackedBubble/PackedBubblePoint.ts:160]: Any tip on how I can avoid this 😬 ? Removing the omit-statement breaks the build, as does removing the extend-keyword entirely 🤔

**Comment by MarkusBarstad** [comment]
Hey @bre1470 , I found additional issues with my original improvement but I think it should work now.

Ive moved the new logic to `BubbleSeries.ts` to avoid making `Series.ts` bigger.

File: ts/Series/PackedBubble/PackedBubblePoint.ts (hunk 3 above)
Comment (bre1470) [reply ts/Series/PackedBubble/PackedBubblePoint.ts:160]: It is not compatible, so extends is misleading. Instead you can reference supported properties as type via `property: TheInterface['theproperty'];` in your declaration. Or you can explicitly omit a property by setting it to undefined, but that should not be done for functions, instead assign the `noop` function from globals in the implementation.

Comment (MarkusBarstad) [reply ts/Series/PackedBubble/PackedBubblePoint.ts]: turns out we did not need to mess with typing after all, there was some residue code which did not matter 😆

**Review by bre1470 — APPROVED** [review]
Looks good! Though I wonder if you could adjust arguments in a way so that you can pass on to `super.searchKDTree` with just a custom `doSearch`. That saves us from redundant code and might help with similar issues in other series.

**Comment by MarkusBarstad** [comment]
> Looks good! Though I wonder if you could adjust arguments in a way so that you can pass on to `super.searchKDTree` with just a custom `doSearch`. That saves us from redundant code and might help with similar issues in other series.

@bre1470 An issue with this is that both implementations of `doSearch` relies upon many variables defined in the original outer function, so that we will have to define those for `BubbleSeries.ts` anyhow 🤔 

I think it is possible to take a smaller portion of the function and refactor it though, i'll give it a shot

**Review by TorsteinHonsi — APPROVED** [review]
Thanks, looks great 👍

**Pull request merged**